### PR TITLE
Adds ability to cluster nearby shows

### PR DESCRIPTION
**Pull request by l2succes**
- Allows us to update the drawer position which is needed to collapse the drawer when the user taps a pin
- Converts shows data in GeoJSON format for consumption by the clustering engine

As of now, `react-native-mapbox-gl` doesn't expose an API to get points inside a cluster. I've attempted using supercluster, the underlying clustering library, to get the data but I still haven't gotten it to return the proper data. 

What we do have is the cluster coordinates and it's size, so another potential solution is sort coordinates by distance and select the N first points which will correspond to the points in the cluster. The downside of this solution is it would become more inefficient as the data size grows.

TODO:
- Handle tapping on a cluster which should show the following UI: https://zpl.io/29oy5Bw

**Screengrabs**

*Clustering Demo*

*Collapsing drawer*

#skip_new_tests

### CODE REVIEW

File: src/lib/Scenes/Map/GlobalMap.tsx [hunk 1]
```diff
@@ -230,6 +257,19 @@ export class GlobalMap extends React.Component<Props, State> {
               trackUserLocation: true,
             })
           }}
+          onPress={async event => {
+            this.setState({
+              activeShowID: null,
+            })
+            const { screenPointX, screenPointY } = event.properties
```
Comment (l2succes) [inline src/lib/Scenes/Map/GlobalMap.tsx]: The idea here is we use user's tap x and y values and the map zoom level to fetch the cluster from our cluster engine at which we map the points back to the shows

File: src/lib/Scenes/Map/GlobalMap.tsx [hunk 2]
```diff
@@ -47,10 +60,16 @@ const ButtonAnimation = {
   },
 }
 
+enum DrawerPosition {
+  open = "open",
+  closed = "closed",
+  collapsed = "collapsed",
+  partiallyRevealed = "partiallyRevealed",
+}
```
Comment (ashfurrow) [inline src/lib/Scenes/Map/GlobalMap.tsx]: Shame that TypeScript can't infer the underly enum values à la Swift.

File: src/lib/Scenes/Map/GlobalMap.tsx [hunk 3]
```diff
@@ -230,6 +257,19 @@ export class GlobalMap extends React.Component<Props, State> {
...
+              activeShowID: null,
+            })
+            const { screenPointX, screenPointY } = event.properties
+            const screenCoords = [screenPointX, screenPointY]
+            const features = await this.map.queryRenderedFeaturesAtPoint(screenCoords, ["==", "type", "Cluster"])
+            console.log(features)
+            const zoom = await this.map.getZoom()
+            const [x, y] = await this.map.getCoordinateFromView(screenCoords)
+            const data = this.clusterEngine.getTile(zoom, x, y)
+            console.log(data)
```
Comment (ashfurrow) [inline src/lib/Scenes/Map/GlobalMap.tsx]: Looks like a few `console.log` calls here.

File: src/lib/utils/convertCityToGeoJSON.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    features: data.map(({ node }) => {
+      const {
+        coordinates: { lat, lng },
+      } = node.location
+
+      return {
+        type: "Feature",
+        geometry: {
+          type: "Point",
+          coordinates: [lng, lat],
```
Comment (ashfurrow) [inline src/lib/utils/convertCityToGeoJSON.ts]: This is long/lat, which appears to be correct GeoJSON, but our codebase uses lat/long. I'd recommend including a comment so someone doesn't come in later and "fix" this 😄

Comment (orta) [reply src/lib/utils/convertCityToGeoJSON.ts]: so weird, but yeah - GeoJSON spec denotes the other way around. Weird

Comment (anandaroop) [reply src/lib/utils/convertCityToGeoJSON.ts]: Yeah this will always be weird until the end of time 

https://macwright.org/lonlat/

😭

Comment (l2succes) [reply src/lib/utils/convertCityToGeoJSON.ts]: lol yeah I was getting confused just working on this, it's even more inconsistent than I thought lol

File: src/lib/Scenes/Map/GlobalMap.tsx (hunk 2 above)
Comment (l2succes) [reply src/lib/Scenes/Map/GlobalMap.tsx]: Yup, that would be nice indeed

**Comment by l2succes** [comment]
Alright thanks @ashfurrow, I addressed your comments. Gonna merge this one

**Pull request merged**